### PR TITLE
Updated input structure

### PR DESCRIPTION
**Pull request by khtufteland**
Solves #28, #31, and #32.
commit df6a1e7963acca415052a0ec57c1071921d6098a
Author: Kjetil Hope Tufteland <kjetil@searis.no>
Date:   Mon Jun 7 15:10:28 2021 +0200

    Added credentials check on input flow

commit f2c8ab0d670e64e478e76fa09d53949201d48c95
Author: Kjetil Hope Tufteland <kjetil@searis.no>
Date:   Fri May 21 05:29:42 2021 +0200

    Updated input structure

### CODE REVIEW

**Review by smyrman — APPROVED** [review]
LGTM

File: nodes/clarify_util.js [hunk 1]
```diff
@@ -32,39 +32,38 @@ module.exports = {
...
-    if (!Array.isArray(times)) {
-      throw 'dataTimes must be array';
+    if (typeof payload !== 'object') {
+      throw 'msg.payload must be object';
     }
 
-    if (!Array.isArray(series)) {
-      throw 'dataSeries must be array';
+    if (_.isEmpty(payload)) {
+      throw 'msg.payload can not be empty';
```
Comment (smyrman) [inline nodes/clarify_util.js:46]: Actually we should allow messages to not have a payload, right? I.e. to only update meta-data....

**Review by smyrman — COMMENTED** [review]
Actually...

File: nodes/locales/en-US/clarify_insert.html [hunk 2]
```diff
@@ -13,67 +13,22 @@ <h3>General Properties</h3>
...
+      <code>times: ["&lt;timestamp&gt;", ...]</code>
+      <code>values: [(&lt;number&gt; || null), ...]</code>
+      If omitted will no data metrics be sent to Clarify.
+    </dd>
+
+    <dt class="optional">signal <span class="property-type">object&lt;Signal&gt; || undefined</span></dt>
+    <dd>
+      Meta data for the signal. Matches <a href="https://docs.clarify.us/reference#signal">signal object in Clarify documentation</a>.
+      If omitted will no meta data be changed in Clarify.
+    </dd>
```
Comment (smyrman) [inline nodes/locales/en-US/clarify_insert.html:31]: I guess there is no way to view the rendered documentation here?

File: nodes/locales/en-US/clarify_insert.html [hunk 3]
```diff
@@ -13,67 +13,22 @@ <h3>General Properties</h3>
...
-    <dt>Data series<span class="property-type">msg.payload.data.series</span></dt>
-    <dd>Values of the data. The length of the array must be equal to Data times</dd>
-    <dd>Expects: array</dd>
-    <dd>Example: <code>[1, 10]</code></dd>
+    <dt class="optional">payload <span class="property-type">object || undefined</span></dt>
+    <dd>
+      Object containing two elements; times and values.
+      <code>times: ["&lt;timestamp&gt;", ...]</code>
+      <code>values: [(&lt;number&gt; || null), ...]</code>
+      If omitted will no data metrics be sent to Clarify.
```
Comment (smyrman) [inline nodes/locales/en-US/clarify_insert.html:24]:
```suggestion
      If omitted, no data will be sent to Clarify.
```

File: nodes/locales/en-US/clarify_insert.html [hunk 4]
```diff
@@ -13,67 +13,22 @@ <h3>General Properties</h3>
...
+    </dd>
+
+    <dt class="optional">signal <span class="property-type">object&lt;Signal&gt; || undefined</span></dt>
+    <dd>
+      Meta data for the signal. Matches <a href="https://docs.clarify.us/reference#signal">signal object in Clarify documentation</a>.
```
Comment (smyrman) [inline nodes/locales/en-US/clarify_insert.html:29]:
```suggestion
      Metadata for the signal. Content must match the <a href="https://docs.clarify.us/reference#signal">Signal object</a> in the Clarify documentation.
```

File: nodes/locales/en-US/clarify_insert.html [hunk 5]
```diff
@@ -13,67 +13,22 @@ <h3>General Properties</h3>
...
+      If omitted will no meta data be changed in Clarify.
```
Comment (smyrman) [inline nodes/locales/en-US/clarify_insert.html:30]:
```suggestion
      If omitted, no meta data will be changed in Clarify.
```

File: nodes/locales/en-US/clarify_insert.html (hunk 2 above)
Comment (khtufteland) [reply nodes/locales/en-US/clarify_insert.html:31]: It's quite easy. Clone [searis/node-red-dev-env](https://github.com/searis/node-red-dev-env), and this repo in an adjacent folder. While in the `node-red-dev-env` folder, run:
```sh
$ npm run watch
```
and access `http://localhost:1880/node-red`.

Is documented in README.md at [searis/node-red-dev-env](https://github.com/searis/node-red-dev-env).

**Pull request merged**